src/wechaty/user/friendship.py

```diff
@@ -138,33 +134,31 @@ def __str__(self) -> str:
             # TODO -> get constructor name of the friendship
             return 'Friendship'
         return 'Friendship # {0} <{1}>'.format(
-            str(self._payload.type),
-            self.payload.contact_id)
+            str(self._payload.type), self._payload.contact_id)
 
     def is_ready(self) -> bool:
         """
         check if friendship is ready
         """
-        return self.puppet is None or self.payload is None
+        return self.puppet is not None and self.payload is not None
```

Comment:
I have a overall consideration about the `ready` and `payload` structure, because the null type checking in python is not as flexable as Typescript, so I think we may consider to make more changes. See #97 Issue.

Reply:
I think you can hold a python-wechaty meeting to discuss these problems and existing problems in `python-wechaty`.